### DISCUSSION TITLE
mongodb-memory-server version for mongodb 6.0.14

### DISCUSSION THREAD

**Discussion by vibsvue** [typegoose/mongodb-memory-server#880] [Q&A]
Hello,
My existing mongodb version 5.0 has been upgraded to 6.0.14. Subsequently, I upgraded the compatible Mongoose version to 6.5.0 from existing- 5.13.14. All the changes done in package.json.
Can you please let me know if there is any equivalent compatible mongodb-memory-server version which need to be updated.
The existing version in package.json is as below-
```json
  "config": {
    "mongodbMemoryServer": {
      "version": "4.4.6"
    }
 "devDependencies": {    
"mongodb-memory-server": "^6.9.3"}}
```
Issue: After the mongoose update, my unit-test job is failing in the pipeline. The log says that- 
```
MongoMS:MongoMemoryServer  - `start()` command was successfully resolved. in-memory-mongo-db-uri:  mongodb://127.0.0.1:. ................
```

**Comment by hasezoey** [comment]
>  My existing mongodb version 5.0 has been upgraded to 6.0.14. Subsequently, I upgraded the compatible Mongoose version to 6.5.0 from existing- 5.13.14. All the changes done in package.json.

do you mean your production database version? because the one defined in the snippet from the package.json indicates a mongodb version of `4.4.6` (not `5.0.0` or `6.0.14`)

>  Can you please let me know if there is any equivalent compatible mongodb-memory-server version which need to be updated.

the latest mongodb-memory-server basically supports all mongodb versions above `3.6.0`, but those versions dont exist for most modern distros anymore, so you should look at [mongodb release archive](https://www.mongodb.com/download-center/community/releases/archive) to get which versions are available for what distro, aside from that i recommend to run the latest possible `mongodb-memory-server` that your project can afford (like node version constraints). Additionally also consult our [Supported Systems](https://nodkz.github.io/mongodb-memory-server/docs/guides/supported-systems/) docs.
(there are currently no specific docs on what MMS versions support what mongodb server range)

TL;DR: use the latest `mongodb-memory-server` package version that you can

**Reply by vibsvue** [reply]
Thank you for sharing this. I will go through it. Just one doubt- That package.json I shared is my dev env. only. So is that mongodbMemoryServer version my mongodb version? I thought this is some memory server which runs locally t o execute unit test cases.
Mongodb database version was upgraded separately by the dbas.

**Reply by hasezoey** [reply]
> So is that mongodbMemoryServer version my mongodb version?

yes that sets your mongodb server binary version to use, see [Config option `VERSION`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#version) (though note that the documentation is for the latest version, not 6.x)

>  I thought this is some memory server which runs locally t o execute unit test cases.

`mongodb-memory-server` uses actual mongodb server binaries, like you would download & start manually. The only difference is that the binary(instance) is managed in a way to easily integrate into tests (or quick repl playing), without having to worry about manually downloading binaries in your CI and asking each developer on your project to locate & download and start a instance and later upgrade that version, in addition this also keeps everyone on the same page regarding mongodb server version

**Reply by vibsvue** [reply]
Thank you
